pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,29 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label less than 63 chars from a given character
+// set, and total length less than 253 chars.  Trailing dots are allowed.
```

Comment:
Both limits are "less than or equal to", aren't they?  
https://github.com/kubernetes/kubernetes/blob/6715318ee738668db0ddb84415f6926208761b9b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L231
https://github.com/kubernetes/kubernetes/blob/6715318ee738668db0ddb84415f6926208761b9b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L210
```suggestion
// at least two labels, with each label no more than 63 chars from a given character
// set, and total length no more than 253 chars.  Trailing dots are allowed.
```